### PR TITLE
FOGL-7980 Initial performance counter implemetation

### PR DESCRIPTION
**Pull request by MarkRiddoch**

### CODE REVIEW

File: C/services/south/ingest.cpp [hunk 1]
```diff
@@ -411,10 +413,13 @@ vector<Reading *> *fullQueue = 0;
 	}
 	if (m_fullQueues.size())
 		m_cv.notify_all();
+	m_performance->collect("queueLegnth", (long)queueLength());
```
Comment (RVerhoeff) [inline C/services/south/ingest.cpp:416]: Typo

File: C/services/south/ingest.cpp [hunk 2]
```diff
@@ -452,14 +457,16 @@ unsigned int nFullQueues = 0;
 	{
 		m_cv.notify_all();
 	}
+	m_performance->collect("queueLegnth", (long)queueLength());
```
Comment (RVerhoeff) [inline C/services/south/ingest.cpp:460]: Typo

File: C/services/south/ingest.cpp [hunk 3]
```diff
@@ -452,14 +457,16 @@ unsigned int nFullQueues = 0;
...
+	m_performance->collect("ingestCount", (long)vec->size());
 }
 
 /**
  * Work out how long to wait based on age of oldest queued reading
- * We do this in a seperaste function so that we can
- * lock the qMutex to access the oldest element in the queue
+ * We do this in a seperate function so that we can lock the qMutex
```
Comment (RVerhoeff) [inline C/services/south/ingest.cpp:466]: Typo: separate

**Review by RVerhoeff — CHANGES_REQUESTED** [review]
Typos

**Review by pintomax — DISMISSED** [review]

**Comment by MarkRiddoch** [comment]
> Typos

It's funny how you read what you think you have written!

**Review by RVerhoeff — APPROVED** [review]

**Pull request merged**